src/renderer/widgets/Unstake/model/form-model.ts

```diff
@@ -0,0 +1,690 @@
+      const address = toAddress(shard.accountId, { prefix: network.chain.addressPrefix });
+      const leftAmount = new BN(staking?.[address]?.active || '0').sub(new BN(amount));
+      const chill = leftAmount.lte(new BN(minBond));
+
+      return transactionBuilder.buildUnstake({
+        chain: network.chain,
+        asset: network.asset,
+        accountId: shard.accountId,
+        amount: form.amount || '0',
+        chill,
+      });
+    });
+  },
+  { skipVoid: false },
+);
+
+const $transactions = combine(
+  {
+    apis: networkModel.$apis,
+    networkStore: $networkStore,
+    pureTxs: $pureTxs,
+    txWrappers: $txWrappers,
+  },
+  ({ apis, networkStore, pureTxs, txWrappers }) => {
+    if (!networkStore || !pureTxs) return undefined;
+
+    return pureTxs.map((tx) =>
+      transactionService.getWrappedTransaction({
+        api: apis[networkStore.chain.chainId],
+        addressPrefix: networkStore.chain.addressPrefix,
+        transaction: tx,
+        txWrappers,
+      }),
+    );
+  },
+  { skipVoid: false },
+);
+
+const $canSubmit = combine(
+  {
+    isFormValid: $unstakeForm.$isValid,
+    isFeeLoading: $isFeeLoading,
+    isStakingLoading: subscribeStakingFx.pending,
+  },
+  ({ isFormValid, isFeeLoading, isStakingLoading }) => {
+    return isFormValid && !isFeeLoading && !isStakingLoading;
+  },
+);
+
+// Fields connections
+
+sample({
+  clock: formInitiated,
+  target: [$unstakeForm.reset, $selectedSignatories.reinit],
+});
+
+sample({
+  clock: formInitiated,
+  filter: ({ chain, shards }) => Boolean(getRelaychainAsset(chain.assets)) && shards.length > 0,
+  fn: ({ chain, shards }) => ({
+    shards,
+    networkStore: { chain, asset: getRelaychainAsset(chain.assets)! },
+  }),
+  target: spread({
+    shards: $shards,
+    networkStore: $networkStore,
+  }),
+});
+
+sample({
+  clock: formInitiated,
+  source: $api,
+  filter: (api): api is ApiPromise => Boolean(api),
+  target: getMinNominatorBondFx,
+});
+
+sample({
+  clock: getMinNominatorBondFx.doneData,
+  target: $minBond,
+});
+
+sample({
+  clock: formInitiated,
+  source: {
+    networkStore: $networkStore,
+    api: $api,
+    shards: $shards,
+  },
+  filter: ({ networkStore, api }) => {
+    return Boolean(networkStore) && Boolean(api);
+  },
+  fn: ({ networkStore, api, shards }) => {
+    const addresses = shards.map((shard) => toAddress(shard.accountId, { prefix: networkStore!.chain.addressPrefix }));
+
+    return {
+      chainId: networkStore!.chain.chainId,
+      api: api!,
+      addresses,
+    };
+  },
+  target: subscribeStakingFx,
+});
+
+sample({
+  clock: subscribeStakingFx.doneData,
+  target: $stakingUnsub,
+});
+
+sample({
+  source: {
+    staking: $staking,
+    networkStore: $networkStore,
+    shards: $unstakeForm.fields.shards.$value,
+  },
+  filter: ({ staking, networkStore }) => Boolean(staking) && Boolean(networkStore),
+  fn: ({ staking, networkStore, shards }) => {
+    if (shards.length === 0) return '0';
```

Comment:
We have `'0'` three times in the function and some in other places, maybe add constant for it?